lib/chain.js

```diff
@@ -157,19 +157,38 @@ function call(handler, err, req, res, _next) {
         _next(nextErr, req, res);
     }
 
+    function resolve() {
+        next();
+    }
+
+    function reject(error) {
+        if (!error) {
+            error = new errors.VError(
+                {
+                    name: 'AsyncHandlerRejection',
+                    info: { handler: handler._name }
+                },
+                `Async middleware ${handler._name} rejected without an error`
+            );
+        }
+        next(error);
```

Comment:
I think it's expected with async-await but this will catch syntax errors etc. 
Maybe explain in the 9.x migration guide.

Reply:
Should the migration guide be part of a separate PR, with the docs?